lib/Artifacts.php

```diff
@@ -0,0 +1,127 @@
+		$checks = $this->client->repo()->checkRuns()->allForReference($this->owner, $this->repo, $ref);
+
+		// Find a check run corresponding to the GitHub Actions build workflow
+		foreach ($checks['check_runs'] as $run) {
+			if ($run['app']['slug'] === 'github-actions' && $run['name'] === 'linux') {
+				$jobId = $run['id'];
+			}
+		}
+		if (!isset($jobId)) { return []; }
+
+		// Get the GitHub Actions workflow run corresponding to that check run
+		$job = $this->client->repo()->workflowJobs()->show($this->owner, $this->repo, $jobId);
+		$runId = $job['run_id'];
+
+		// Get all artifacts of that workflow run
+		$artifacts = $this->client->repo()->artifacts()->runArtifacts($this->owner, $this->repo, $runId);
+
+		$validArtifacts = array_filter($artifacts['artifacts'], function ($artifact) {
+			return !$artifact['expired'];
+		});
+		$description = '## ' . $pr['title'] . "\n" . $pr['body'];
+		return $this->mapPullRequestAssetsFromJson($validArtifacts, $id, $description);
+	}
+
+	public function getArtifactDownloadUrl(int $artifactId): string
+	{
+		return (string) $this->client->repo()->artifacts()->download($this->owner, $this->repo, $artifactId);
+	}
+
+	private function mapBranchAssetsFromJson(array $json): array
+	{
+		return array_map(function (array $artifact) {
+			return new Asset(
+				platform: self::platformFromArtifactName($artifact['name']),
+				platformName: self::platformNameFromArtifactName($artifact['name']),
+				releaseName: 'g' . substr($artifact['workflow_run']['head_sha'], 0, 9),
+				downloadUrl: $this->router->generate('download_artifact', ['id' => $artifact['id']]),
+				description: null,
+				gitRef: $artifact['workflow_run']['head_sha'],
+				date: $artifact['created_at']
+			);
+		}, $json);
+	}
+
+	private function mapPullRequestAssetsFromJson(array $json, string $pr, string $description): array
+	{
+		return array_map(function (array $artifact) use ($pr, $description) {
+			return new Asset(
+				platform: self::platformFromArtifactName($artifact['name']),
+				platformName: self::platformNameFromArtifactName($artifact['name']),
+				releaseName: '#' . $pr . '@' . substr($artifact['workflow_run']['head_sha'], 0, 9),
+				downloadUrl: $this->router->generate('download_artifact', ['id' => $artifact['id']]),
+				description: $description,
+				gitRef: $artifact['workflow_run']['head_sha'],
+				date: $artifact['created_at']
+			);
+		}, $json);
+	}
+
+	private static function platformFromArtifactName(string $artifactName): Platform
+	{
+		switch ($artifactName) {
+			case 'linux': return Platform::Linux;
+			case 'macos': return Platform::MacOS;
+			case 'mingw32': return Platform::Windows;
+			case 'mingw64': return Platform::Windows;
+			default: return Platform::Unknown;
+		}
+	}
+
+	private static function platformNameFromArtifactName(string $artifactName): string
+	{
+		switch($artifactName) {
+			case 'linux': return 'Linux 64-bit';
+			case 'macos': return 'macOS 10.15+';
```

Comment:
Not in the website code alone, unfortunately. For releases, we get the version from the dmg file name, but that's not possible here without downloading the whole artifact, since the `upload-artifact` puts it in a zip file. We could try and change the artifact name to better reflect the file it contains, or perhaps generate a lightweight metadata artifact that we can download to get more information about a build.

Reply:
A possible alternative is parsing the action `.yml` file, whose path can be obtained from the action runs API. I think it sounds too complicated, however, so you may not want to do that.
BTW, I think we use macOS 11 images for GitHub Actions now(see https://github.com/LMMS/lmms/blob/17aa7dbef5f4b1dbf8c7c335ef6f2a7759600f21/.github/workflows/build.yml#L65-L67).

Reply:
We use Xcode 11.7 in that image though, which targets macOS 10.15: https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md#installed-sdks. If we wanted to parse the action file, either for the macOS version or the Xcode version, we'd still have to consult a mapping from Xcode version to macOS SDK. I agree it's probably too complicated, and it sounds rather fragile too.

Reply:
> We use Xcode 11.7 in that image though, which targets macOS 10.15

Oh yes, I missed the `DEVELOPER_DIR` part.

> we'd still have to consult a mapping from Xcode version to macOS SDK.

Uh, it doesn't sound good...